storage/storage.go

```diff
@@ -0,0 +1,122 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"github.com/gazebo-web/gz-go/v7"
+	"github.com/pkg/errors"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+// Storage holds the methods to interact with a Cloud provider storage.
+type Storage interface {
+	// GetFile returns the content of file from a given path.
+	GetFile(ctx context.Context, resource Resource, path string) ([]byte, error)
+	// Download returns a URL to download a resource from.
+	Download(ctx context.Context, resource Resource) (string, error)
+	// UploadDir uploads assets located in the given source folder and placed them into the given resource.
+	UploadDir(ctx context.Context, resource Resource, source string) error
+	// UploadZip uploads a compressed set of assets of the given resource.
+	UploadZip(ctx context.Context, resource Resource, file *os.File) error
```

Comment:
Sorry for coming back to the interface definition after making you change the implementations so much, but after reviewing and checking how this interface is used in other projects, this method should probably be renamed to `UploadFile`. 

It is currently called `UploadZip` because it is used to upload a zip file inside of fuel-server, but it also expects the provided file to already be zipped, making the `Zip` part not really a requirement. Since this is merely a rename of the method, I'd like to make this change in this PR before merging.